CMakeLists.txt

```diff
@@ -107,6 +107,8 @@ endif()
 option(PYBIND11_INSTALL "Install pybind11 header files?" ${PYBIND11_MASTER_PROJECT})
 option(PYBIND11_TEST "Build pybind11 test suite?" ${PYBIND11_MASTER_PROJECT})
 option(PYBIND11_NOPYTHON "Disable search for Python" OFF)
+option(PYBIND11_DISABLE_HANDLE_TYPE_NAME_DEFAULT_IMPLEMENTATION
+       "To enforce that a handle_type_name<> specialization exists" OFF)
```

Comment:
Should we indicate that this is an internal option, and maybe even hide it? Users aren't expected to use this or rely on it existing.

Reply:
I could be useful, for others to discover missing specializations, e.g. if I wanted to clean up the Google codebase:

https://github.com/pybind/pybind11/pull/4888#issuecomment-1871790589

Ultimately these are real bugs that erode type checking completeness.

I wouldn't go as far as prominently advertising the option (I figure most people don't have time to clean up things like this), but I also wouldn't want to hide it.

Reply:
Okay.